### PR TITLE
Chat plugins: support a way to set custom avatars

### PR DESCRIPTION
**Pull request by mia-pi-git**
Continuation of #5515, but i deleted my old fork, so i can't just use that PR.
Note for those looking that this is just for side servers.

### CODE REVIEW

**Comment by Zarel** [comment]
I thought this was drafted, but it's possible that it was finished sometime between our incredibly busy phase where I didn't have time to look at it. In any case, it could at least use an update to `Net`.

File: server/chat-plugins/customavatars.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+import {FS} from '../../lib/fs';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as https from 'https';
+import * as http from 'http';
+import * as url from 'url';
+
+
+export const avatarManager = new class {
```
Comment (Zarel) [inline server/chat-plugins/customavatars.ts:14]:
```suggestion
export const AvatarManager = new class {
```

**Comment by DaWoblefet** [comment]
Superseded by https://github.com/smogon/pokemon-showdown/pull/8563 which is live.

**Comment by Zarel** [comment]
Not completely superseded. There's one more feature I'd expect side servers to want, which is a way to grab an avatar from a URL and put it in `config/avatars/`.

**Comment by Zarel** [comment]
Which this code _does_ provide!

**Pull request closed without merging**